api-client/src/main/java/com/desk/java/apiclient/util/OpportunityActivityAdapter.java

```diff
@@ -88,8 +90,10 @@ public IOpportunityActivity deserialize(JsonElement jsonElement, Type type, Json
                     if (HISTORY.equalsIgnoreCase(clazz.getAsString())) {
                         return gson.fromJson(jsonElement, OpportunitySystemEvent.class);
 
-                    // if the class is of type 'opportunity_activity' then it's an opportunity activity
-                    } else if (OPPORTUNITY_ACTIVITY.equalsIgnoreCase(clazz.getAsString())) {
+                    // if the class is of type 'opportunity_activity' or 'attachment'
+                    } else if (OPPORTUNITY_ACTIVITY.equalsIgnoreCase(clazz.getAsString()) ||
+                            OPPORTUNITY_ATTACHMENT.equalsIgnoreCase(clazz.getAsString())) {
+
                         JsonPrimitive activityType = element.getAsJsonPrimitive(TYPE);
 
                         if (OpportunityActivityTypes.CALL.equalsIgnoreCase(activityType.getAsString())) {
```

Comment:
If the activity type field is changed to be the enum then we could just have a switch here to check the type.